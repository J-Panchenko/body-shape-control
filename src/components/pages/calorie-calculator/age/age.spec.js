import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import Age from './age';

const setup = () => {
  const utils = render(<Age />);
  const input = utils.getByLabelText('age-input');
  return {
    input,
    ...utils,
  };
};

describe('Age component: positive cases', () => {
  const { input } = setup();

  it('It should allow numbers to be inputted', () => {
    fireEvent.change(input, { target: { value: '87' } });
    expect(input.value).toBe('87');
  });

  it('It should not allow letters to be inputted', () => {
    fireEvent.change(input, { target: { value: 'Good Day' } });
    expect(input.value).toBe('');
  });

  it('It should allow numbers from 15 to 100 to be inputted', () => {
    fireEvent.change(input, { target: { value: '102' } });
    expect(input.value).toBe('102');
    expect(input.validity.stepMismatch).toBe(false);

    fireEvent.change(input, { target: { value: '12' } });
    expect(input.value).toBe('12');
    expect(input.validity.stepMismatch).toBe(false);
  });
});
