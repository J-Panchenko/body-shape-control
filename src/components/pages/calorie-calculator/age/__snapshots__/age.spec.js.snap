// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age component: positive cases It should be renders correctly 1`] = `
<label
  className="parameter__item"
  htmlFor="age"
>
  <h2
    className="parameter-subtitle"
  >
    Age
  </h2>
  <span>
    years
  </span>
  <input
    aria-label="age-input"
    className="input"
    id="age"
    max={100}
    min={15}
    placeholder="0"
    required={true}
    step={1}
    type="number"
  />
</label>
`;
